### ISSUE TITLE
panicked at 'attempt to subtract with overflow', src/lib.rs:129:20

### ISSUE THREAD

**Issue by purew** [georust/polyline#27]
I have a large `polyline6` which seems to trigger an overflow in the decoder. It's recreated with the following patch
[overflow.patch.txt](https://github.com/georust/polyline/files/10387367/overflow.patch.txt)

`cargo test` then produces

```
test tests::should_not_trigger_overflow ... FAILED

failures:

---- tests::should_not_trigger_overflow stdout ----
thread 'tests::should_not_trigger_overflow' panicked at 'attempt to subtract with overflow', src/lib.rs:129:20


failures:
    tests::should_not_trigger_overflow

test result: FAILED. 4 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 7.76s

error: test failed, to rerun pass `--lib````

**Comment by purew** [comment]
Actually, I've figured out that the polyline contains a line-feed character as the last byte.

The python polyline library I compared against ignores this line-feed character while this library produces the above-mentioned out-of-bounds error.

I guess this technically is invalid input, but as other libraries don't misbehave badly with such input, I'll leave it up to the maintainer to classify this behaviour as a bug or if the ticket should be closed with no action.
